environment/k8s_environment.go

```diff
@@ -559,7 +567,87 @@ func (m *K8sManifest) ServiceDetails() ([]*ServiceDetails, error) {
 	return serviceDetails, nil
 }
 
-// Values returns all the values to be exposed in the definition templates
+func (m *K8sManifest) GetPodsFullNames(partialName string) ([]string, error) {
```

Comment:
When the deployment forwards ports, it already has access to all of the pod names so rather than having a function just to get pod names, store the pod name with the port information on port forwarding.

Sergey did this with his Helm PR where he wrapped the forwarded ports into its own struct:
https://github.com/smartcontractkit/integrations-framework/blob/main/environment/k8s_helm_resource.go#L30

Use the same pattern to store the pod name.